web_drop_target/static/src/js/web_drop_target.js

```diff
@@ -27,44 +28,67 @@ odoo.define('web_drop_target', function(require) {
         },
 
         _on_drop: function(e) {
-            var drop_item = this._get_drop_item(e);
-            if(!drop_item) {
-                return;
-            }
-            jQuery(e.delegateTarget).removeClass(this._drag_over_class);
-            var reader = new FileReader();
-            reader.onloadend = this.proxy(
-                _.partial(this._handle_file_drop, drop_item.getAsFile())
-            );
-            reader.readAsArrayBuffer(drop_item.getAsFile());
+            var self = this;
+            var drop_items = this._get_drop_item(e);
+            _.each(drop_items, function(drop_item) {
+                var reader = new FileReader();
+                reader.onloadend = self.proxy(
+                    _.partial(self._handle_file_drop, drop_item.getAsFile())
+                );
+                reader.readAsArrayBuffer(drop_item.getAsFile());
+            });
+            this._drop_overlay.remove();
+            this._drop_overlay = null;
             e.preventDefault();
         },
 
         _on_dragenter: function(e) {
-            if(this._get_drop_item(e)) {
+            if(this._get_drop_item(e).length) {
                 e.preventDefault();
-                jQuery(e.delegateTarget).addClass(this._drag_over_class);
+                if(!this._drop_overlay){
+                    var drop_overlay_message = _t('Drop your files here');
+                    var o_content = jQuery('.o_content');
+                    var view_manager = jQuery('.o_view_manager_content');
+                    this._drop_overlay = jQuery(
+                        `<div class="o_drag_over">
+                            <div class="o_drag_over_content">
+                                <div><i class="fa fa-file-o fa-5x" aria-hidden="true"></i></div>
+                                <div><h2>${drop_overlay_message}</h2></div>
+                            </div>
+                        </div>`
+                    );
+                    var o_content_position = o_content.position();
+                    this._drop_overlay.css({
+                        'top': o_content_position.top, 
+                        'left': o_content_position.left,
+                        'width': view_manager.width(),
+                        'height': view_manager.height()
+                    });
+                    o_content.append(this._drop_overlay);
+                }
                 return false;
             }
         },
 
         _on_dragleave: function(e) {
-            jQuery(e.delegateTarget).removeClass(this._drag_over_class);
+            this._drop_overlay.remove();
+            this._drop_overlay = null;
```

Comment:
let's refactor 'do whatever we need to create/remove the d&d ui' into their own functions for simpler adaptions

Reply:
Done